From 7d66bb5fa2710ad2cc12a5c067fe9081d2135e7e Mon Sep 17 00:00:00 2001
From: brendan King <Brendan.King@imgtec.com>
Date: Thu, 16 Sep 2021 17:46:28 +0100
Subject: [PATCH 11/12] mesa: partially revert pbuffer attribute removal

This partially reverts commit 5ffd1ebe6b3c8c7dd316dd47fac088044222e6ef
("mesa: Remove misc pbuffer attributes from struct gl_config").

The IMG PowerVR driver sets meaningful values for the maximum
pbuffer width, height and pixels.
---
 src/mesa/drivers/dri/common/utils.c | 7 ++++---
 src/mesa/main/mtypes.h              | 5 +++++
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/src/mesa/drivers/dri/common/utils.c b/src/mesa/drivers/dri/common/utils.c
index a3f2bc57f46..4f5b7a844fb 100644
--- a/src/mesa/drivers/dri/common/utils.c
+++ b/src/mesa/drivers/dri/common/utils.c
@@ -398,6 +398,7 @@ __DRIconfig **driConcatConfigs(__DRIconfig **a,
     return all;
 }
 
+/* careful, lack of trailing semicolon */
 #define __ATTRIB(attrib, field) case attrib: *value = config->modes.field; break
 
 /**
@@ -469,9 +470,9 @@ driGetConfigAttribIndex(const __DRIconfig *config,
     __ATTRIB(__DRI_ATTRIB_GREEN_MASK,			greenMask);
     __ATTRIB(__DRI_ATTRIB_BLUE_MASK,			blueMask);
     __ATTRIB(__DRI_ATTRIB_ALPHA_MASK,			alphaMask);
-    case __DRI_ATTRIB_MAX_PBUFFER_WIDTH:
-    case __DRI_ATTRIB_MAX_PBUFFER_HEIGHT:
-    case __DRI_ATTRIB_MAX_PBUFFER_PIXELS:
+    __ATTRIB(__DRI_ATTRIB_MAX_PBUFFER_WIDTH,		maxPbufferWidth);
+    __ATTRIB(__DRI_ATTRIB_MAX_PBUFFER_HEIGHT,		maxPbufferHeight);
+    __ATTRIB(__DRI_ATTRIB_MAX_PBUFFER_PIXELS,		maxPbufferPixels);
     case __DRI_ATTRIB_OPTIMAL_PBUFFER_WIDTH:
     case __DRI_ATTRIB_OPTIMAL_PBUFFER_HEIGHT:
     case __DRI_ATTRIB_VISUAL_SELECT_GROUP:
diff --git a/src/mesa/main/mtypes.h b/src/mesa/main/mtypes.h
index 989d652bb15..68d4998c1b1 100644
--- a/src/mesa/main/mtypes.h
+++ b/src/mesa/main/mtypes.h
@@ -174,6 +174,11 @@ struct gl_config
    /* ARB_multisample / SGIS_multisample */
    GLuint samples;
 
+   /* GLX 1.3 */
+   GLint maxPbufferWidth;
+   GLint maxPbufferHeight;
+   GLint maxPbufferPixels;
+
    /* OML_swap_method */
    GLint swapMethod;
 
-- 
2.25.1

